### DISCUSSION TITLE
weapon mods may not give an improvement when stats exceed a threshold

### DISCUSSION THREAD

**Discussion by Anaglyphic** [CleverRaven/Cataclysm-DDA#48907] [General]
Does anyone agree this is wrong, and if so, How should this be handled?

Some of it I _strongly disagree_ with, setting thresholds is not authentic - no matter if you're a professional assassin with the muscle memory of 100s of weapon handling and dozens of kills and thousands of rounds in practice and the dexterity and perception of a god, you will still get some improvements from a laser sight, both speed and dispersion/recoil...

```
    /** If this value is set (non-negative), this gunmod functions as a sight. 
        A sight is only usable to aim by a character whose current 
        @ref Character::recoil is at or below this value. */
    int sight_dispersion = -1;

    /**
     *  For sights (see @ref sight_dispersion), this value affects time cost of aiming.
     *  Higher is better. In case of multiple usable sights,
     *  the one with highest aim speed is used.
     */
```
I quoted sight dispersion because that ones most obvious, take for example rail/underbarrel laser sight, at sight dispersion 30 aim 9. Depending on your stats and base weapon, you might get a nice improvement over the stock, unmodified weapon, a slight improvement, none at all, an sometimes it just makes the weapon heavier buikier and slower.

I don't want to argue about that or badger you to think like me, if it bothers me I know I can change my game to be whatever I want. What I want to discuss is, should there be a basic way to indicate that weapons will not be improved or even made worse by adding things sometimes; some way to know while on the inventory or item summary screen?

As it stands you have to activate and go through as though you're installing it to get the A/B compare screen up, and even then if you don't know what's what or why that laser or holo sight or grip isn't improving anything? Maybe just a pop-up like you get when you try to eat when full?
 
**Installing this does not improve your _\<weapon_id\>_ in any way, but will still increase weight and slow down operation. Install _\<mod_id\>_ anyway? (Case Sensitive) (Y)es  (N)o**

Thoughts?

Further, should there be a "fudge" range for some displays?

It's done for weights and volumes for inventory; you'll see rounding errors - a recipe produce 1x more than it should, or a container take one more/less object than it should, etc. But in object comparisons it's "math-y" precise display. It'll show a weapon base aim speed of weapon A at 375 in green, but weapon B in red at 374. Sure yes it's less and therefore correct, but when you're looking at all stats that affect if it's "better" or "worse" I'm guessing people will only pay attention to the ranged damage value's colors, and if that's the case, what's the damn point of all the values displaying? ^.^ Why not a x% fudge factor for _displaying_ (not computing) values that are, as far as turns and gameplay are concerned. essentially equivalent?  

Again, is it just me? Don't know why it bugs me, I'm hoping typing it out will bug me less.

**Comment by actual-nh** [comment]
Possibly a squared-mean-of-roots system as with engine power addition? That may still cause reductions on adding things, however...

**Comment by actual-nh** [comment]
@Anaglyphic: I suggest opening a bug report issue, specifically focused on the user interface issues. Given likely string changes, it'll need to be after 0.F is out, but I can put a 0.F String Freeze on it to keep it from getting removed as "stale".
